### PR TITLE
Wip/evaluation

### PR DESCRIPTION
**Pull request by OmeGak**

### CODE REVIEW

File: indico/modules/events/surveys/fields/choices.py [hunk 1]
```diff
@@ -73,6 +75,18 @@ def create_wtf_field(self):
...
         return self._make_wtforms_field(field_class, choices=choices, **field_options)
 
+    def get_summary(self, include_empty=False):
+        counter = Counter()
+        for answer in self.question.answers:
+            if answer.data is not None or include_empty:
+                counter[answer.data] += 1
+        total = sum(counter.values())
+        options = self.question.field_data['options']
+        results = {'total': total,
```
Comment (mic4ael) [inline indico/modules/events/surveys/fields/choices.py]: Is it necessary to assign it to a variable?

Comment (ThiefMaster) [inline indico/modules/events/surveys/fields/choices.py]: it's used twice

Comment (ThiefMaster) [inline indico/modules/events/surveys/fields/choices.py]: yeah, i'd return it directly

File: indico/modules/events/surveys/fields/choices.py [hunk 2]
```diff
@@ -136,3 +150,18 @@ def validators(self):
     def wtf_field_kwargs(self):
         return {'choices': [(x['id'], x['option']) for x in self.question.field_data['options']],
                 'coerce': lambda x: x}
+
+    def get_summary(self):
+        counter = Counter()
+        for answer in self.question.answers:
+            for choice in answer.data:
+                counter[choice] += 1
```
Comment (mic4ael) [inline indico/modules/events/surveys/fields/choices.py]: You could use `Counter`'s interface by using `counter.update(choice)`

Comment (ThiefMaster) [inline indico/modules/events/surveys/fields/choices.py]: :+1:, but i think it'd be `counter.update(answer.data)`

Comment (mic4ael) [inline indico/modules/events/surveys/fields/choices.py]: Hm, not really. However, isn't it enough to use it as `Counter(answer.data)`?

Comment (mic4ael) [inline indico/modules/events/surveys/fields/choices.py]: `for answer in self.question.answers:
    counter.update(Counter(answer.data))
`

I am not sure, but maybe this is enough

Comment (ThiefMaster) [inline indico/modules/events/surveys/fields/choices.py]: i think my line already does the job, no need to create a new counter

Comment (mic4ael) [inline indico/modules/events/surveys/fields/choices.py]: Ahh, True ;)

Comment (OmeGak) [inline indico/modules/events/surveys/fields/choices.py]: Nice catch. Didn't know you can pass an iterable.

File: indico/modules/events/surveys/templates/manage_survey.html [hunk 3]
```diff
@@ -38,7 +38,7 @@
                 {% trans %}Submissions{% endtrans %}
             </div>
             <div class="i-box-buttons">
-                <a href="#" class="i-button">
+                <a href="{{ url_for('.survey_results', survey) }}" class="i-button">
                     {% trans %}Show results{% endtrans %}
```
Comment (mic4ael) [inline indico/modules/events/surveys/templates/manage_survey.html]: Could you add some icon to this button? The reason why is that there are two additional buttons for exporting and deleting submissions and both of them have icons.

Comment (OmeGak) [inline indico/modules/events/surveys/templates/manage_survey.html]: I'll add it when your PR is merged. ;)

File: indico/modules/events/surveys/fields/base.py [hunk 4]
```diff
@@ -58,9 +58,30 @@ class SurveyField(object):
     #: the validator to use if the field is not required
     not_required_validator = Optional
 
+    @property
```
Comment (ThiefMaster) [inline indico/modules/events/surveys/fields/base.py]: can you move those back after `__init__`?

File: indico/modules/events/surveys/templates/survey_results.html [hunk 5]
```diff
@@ -5,5 +5,146 @@
...
+                    </div>
+                </div>
+                <div class="i-box-content">
+                    {% if question.field_type == 'text' %}
+                        {{ text_answer(question) }}
+                    {% elif question.field_type == 'number' %}
+                        {{ number_answer(question) }}
+                    {% elif question.field_type in ('bool', 'single_choice') %}
+                        {{ single_choice_answer(question) }}
+                    {% elif question.field_type == 'multiselect' %}
```
Comment (ThiefMaster) [inline indico/modules/events/surveys/templates/survey_results.html]: shouldn't we have an `else` here that uses some generic way to render the answer? Even a plugin could define a new field type...

File: indico/modules/events/surveys/templates/survey_results.html [hunk 6]
```diff
@@ -5,5 +5,146 @@
...
+            donut: true,
+            donutWidth: 20,
+            startAngle: 270,
+            total: 1,
+            chartPadding: 20,
+            labelOffset: 20,
+            labelDirection: 'explode'
+        }).on('draw', function(data) {
+            if (data.value === 0) {
+                chart.data.labels[data.index] = ''
```
Comment (ThiefMaster) [inline indico/modules/events/surveys/templates/survey_results.html]: missing semicolon

File: indico/modules/events/surveys/templates/survey_results.html (hunk 5 above)
Comment (OmeGak) [inline indico/modules/events/surveys/templates/survey_results.html]: `get_summary` raises `NotImplementedError`. Doesn't make much sense to me that here we have a default result summary.

File: indico/modules/events/surveys/templates/survey_results.html (hunk 6 above)
Comment (OmeGak) [inline indico/modules/events/surveys/templates/survey_results.html]: I'm moving the generation of the charts to `surveys.js` and it's already fixed locally.

File: indico/modules/events/surveys/templates/survey_results.html (hunk 5 above)
Comment (ThiefMaster) [inline indico/modules/events/surveys/templates/survey_results.html]: Then maybe the default should be calling a method on the field class so a custom field type can use its own template/macro

Comment (OmeGak) [inline indico/modules/events/surveys/templates/survey_results.html]: I'd extend this to accommodate plugin-defined field types when we need it.

Comment (ThiefMaster) [inline indico/modules/events/surveys/templates/survey_results.html]: in that case let's open an issue. it's one of the easy things new developers can do ;)

**Pull request merged**